mne/io/boxy/boxy.py

```diff
@@ -527,6 +533,90 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
+                        print('Fixing phase wrap')
+                        for i_chan in range(np.size(data_, axis=0)):
+                            if np.mean(data_[i_chan,:50]) < 180:
+                                wrapped_points = data_[i_chan, :] > 270
+                                data_[i_chan, wrapped_points] -= 360
+                            else:
+                                wrapped_points = data_[i_chan,:] < 90
+                                data_[i_chan, wrapped_points] += 360
+                                
+                        # unwrapped_data = scipy.io.loadmat(r"C:\Users\spork\Desktop\data_unwrap_python.mat")
+                        
+                        # test1 = abs(unwrapped_data['data'] - data_) <= thresh
+                        # test1.all()
+                                
+                        print('Detrending phase data')
+                        # scipy.io.savemat(file_name = r"C:\Users\spork\Desktop\data_unwrap_matlab.mat",
+                        #               mdict=dict(data=data_))
+                        
+                        y = np.linspace(0, np.size(data_, axis=1)-1,
+                                        np.size(data_, axis=1))
+                        x = np.transpose(y)
+                        for i_chan in range(np.size(data_, axis=0)):
+                            poly_coeffs = np.polyfit(x,data_[i_chan, :] ,3)
+                            tmp_ph = data_[i_chan, :] - np.polyval(poly_coeffs,x)
+                            data_[i_chan, :] = tmp_ph
+                            
+                        # detrend_data = scipy.io.loadmat(r"C:\Users\spork\Desktop\data_detrend_python.mat")
+                            
+                        # test2 = abs(detrend_data['data'] - data_) <= thresh
+                        # test2.all()
+                        
+                        print('Removing phase mean')
+                        # scipy.io.savemat(file_name = r"C:\Users\spork\Desktop\data_detrend_matlab.mat",
+                        #               mdict=dict(data=data_))
+                        
+                        mrph = np.mean(data_,axis=1);
+                        for i_chan in range(np.size(data_, axis=0)):
+                            data_[i_chan,:]=(data_[i_chan,:]-mrph[i_chan])
+                            
+                        # mean_data = scipy.io.loadmat(r"C:\Users\spork\Desktop\data_mean_python.mat")
+                        
+                        # test3 = abs(mean_data['data'] - data_) <= thresh
+                        # test3.all()
+                        
+                        print('Removing phase outliers')
+                        # scipy.io.savemat(file_name = r"C:\Users\spork\Desktop\data_mean_matlab.mat",
+                        #               mdict=dict(data=data_))
+                        
+                        ph_out_thr=3; # always set to "3" per Kathy & Gabriele Oct 12 2012
```

Comment:
again lets justify this choice of 3, what does this do in english, and why is > 3 an outlier?

Reply:
kk, I'll add a comment explaining this based on the reference provided